src/rollup/types.d.ts

```diff
@@ -495,7 +495,13 @@ export type SourcemapPathTransformOption = (
 	sourcemapPath: string
 ) => string;
 
-export type InputPluginOption = Plugin | null | false | undefined | InputPluginOption[];
+export type InputPluginOption =
+	| Plugin
+	| null
+	| false
+	| undefined
+	| InputPluginOption[]
+	| Promise<Plugin | false | null | undefined | InputPluginOption[]>;
```

Comment:
As a mini optimization so we do not need to repeat everything, maybe use either an intermediate type so that

```ts
type InputPluginOption = InputPluginOptionValue | Promise<InputOptionValue>
```

or introduce a utility type like

```ts
type MaybePromise<T> = T | Promise<T>;
```

that could also be reused in other places?

Reply:
Done. I'll make more PR to improve and simplify types.